packages/opentelemetry-core/src/context/propagation/composite.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Carrier, Context, HttpTextFormat } from '@opentelemetry/api';
+
+/** Combines multiple propagators into a single propagator. */
+export class CompositePropagator implements HttpTextFormat {
+  private _propagators: HttpTextFormat[];
+  constructor(...propagators: HttpTextFormat[]) {
+    this._propagators = propagators;
+  }
+
+  inject(context: Context, carrier: Carrier) {
+    for (const propagator of this._propagators) {
+      try {
+        propagator.inject(context, carrier);
+      } catch {}
```

Comment:
This silently discards exceptions which is not that nice and may result in hard to find bugs. if the same propagator would be used standalone the exception would fall through. I don't think such differences are easy to understand.
I would compare this with node.js eventemitter which also doesn't catch/discard exceptions thrown by listeners

Reply:
I agree silently discarding is probably not the way to go, but do you think we should allow the exception to bubble up, or should we log and continue?

Reply:
Yes, I think it should bubble up. The calling plugin can still decided to handle it.

We may add some logs but as more and more stuff is running in some fast disappearing containers logs are often not accessible when needed.

Reply:
we should log and continue.

Reply:
@mayurkale22 opinion here?

Reply:
I would log (warning) and continue.

Reply:
If the exception doesn't break the client functionality but only opentelemetry I would agree with Flarna that it should not fail silently and error should be raised instead of log.